src/plugins/intel_gpu/include/intel_gpu/primitives/bucketize.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "primitive.hpp"
+
+namespace cldnn {
+/// @addtogroup cpp_api C++ API
+/// @{
+/// @addtogroup cpp_topology Network Topology
+/// @{
+/// @addtogroup cpp_primitives Primitives
+/// @{
+
+/// @brief Bucketize operation bucketizes the input based on boundaries.
+struct bucketize : primitive_base<bucketize> {
+    CLDNN_DECLARE_PRIMITIVE(bucketize)
+
+    /// @brief Constructs bucketize primitive.
+    /// @param id This primitive id.
+    /// @param input Input primitive id.
+    /// @param buckets Sorted unique boundaries for buckets primitive id.
+    /// @param output_type Output tensor type.
+    /// @param with_right_bound Indicates whether bucket includes the right or the left edge of interval.
+    bucketize(const primitive_id& id,
+              const primitive_id& input,
+              const primitive_id& buckets,
+              data_types output_type = data_types::i64,
```

Comment:
output_data_type is already presented in primitive base, isn't it?

Reply:
Yeah, it's there, and I use it a few lines below. But we need somehow to pass this value to output_data_type and this constructor does it.

Reply:
hm, I thought it's initialized somewhere in parent, like inputs

Reply:
Not sure, we have the same logic in "convolution", "concatenation", "arg_max_min", etc.

Reply:
So I can leave this?